### PR TITLE
347 evans pres repr report

### LINKED ISSUE #347: Create a scheduled "Evans Pres Repr" Report Workflow
This will be a monthly report that simply identifies all items with a temporary location of "Evans Pres Repr" and emails the results to a configured email address. It is ok to use the LDP to get this data. This workflow does not require a front end.

The report should be run on a monthly basis, and the configured email address should be sbuckner@library.tamu.edu.

### PR DESCRIPTION
**Pull request by rmathew1011**
Resolves #347

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -570,3 +570,32 @@ curl --location --request POST 'http://localhost:9001/mod-workflow/events/workfl
   --data-raw '{"logLevel": "INFO", "bcn-mail-from": "folio@k1000.library.tamu.edu", "startRange": "a0", "endRange":"b9","username":"*","password":"*", "bcm-mail-to": "recipient@tamu.edu", "path": "/mnt/workflows/diku/bcn" }'
 
 ```
+
+## evans-pres-repr
+
+### Evans Pres Repr Workflow (Scheduled)
+
+This workflows sends out a monthly email a list of all items  with a temporary location of **Eva Pres Repr** for a specific configured email address **evansPresReprFrom**.
```
Comment (kaladay) [inline README.md:578]: Remove extra space and improve grammar.
```suggestion
This workflows sends out a monthly email of a list of all items with a temporary location of **Eva Pres Repr** for a specific configured email address `evansPresReprFrom`.
```

File: evans-pres-repr/nodes/query.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "id": "44ba063b-d801-4afb-a7dd-95b4d5fd1326",
+  "name": "Query items with location as Evans Pres Repr",
+  "description": "Query items with temp loc of Evans Pres Repr",
```
Comment (kaladay) [inline evans-pres-repr/nodes/query.json:4]: Please make this a complete sentence without as many acronyms.
This does not need to be short like the title (name).
I suggest using single quotes to handle the specially named `Evans Pres Repr`.

File: evans-pres-repr/workflow.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "id": "c5b5c214-d06b-4037-888d-27de3d376e11",
+  "name": "Evans Pres Repr Report",
+  "description": "Generate monthly Evans Pres Repr report.",
```
Comment (kaladay) [inline evans-pres-repr/workflow.json:4]: I suggest using single quotes to handle the specially named `Evans Pres Repr`.

File: evans-pres-repr/nodes/email.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "id": "939e1dfc-840f-4261-b8a3-d7a19a32719c",
+  "name": "Send Evans Pres Repr Email",
+  "description": "",
```
Comment (kaladay) [inline evans-pres-repr/nodes/email.json:4]: Add description.

File: evans-pres-repr/nodes/delete.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+  "id": "61dfa92b-e89b-4f69-b16f-84a703f0b33f",
+  "name": "Delete Evans Pres Repr CSV",
+  "description": "",
```
Comment (kaladay) [inline evans-pres-repr/nodes/delete.json:4]: Add description.

File: evans-pres-repr/nodes/connect.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "id": "1d11bca4-87ee-48d7-9b75-3dac8a32e060",
+  "name": "Connect LDP",
+  "description": "",
```
Comment (kaladay) [inline evans-pres-repr/nodes/connect.json:4]: Add description.

File: evans-pres-repr/nodes/email.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      "key": "count",
+      "type": "PROCESS"
+    }
+  ],
+  "outputVariable": {},
+  "mailTo": "{{evansPresReprTo}}",
+  "mailFrom": "{{evansPresReprFrom}}",
+  "mailSubject": "Evans Pres Repr Report",
+  "mailText": "${(count > 0)?then('Monthly Evans Pres Repr CSV file attached', 'No results to report')}",
```
Comment (kaladay) [inline evans-pres-repr/nodes/email.json:24]:
```suggestion
  "mailText": "${(count > 0) ? then('Monthly Evans Pres Repr CSV file attached', 'No results to report')}",
```

File: evans-pres-repr/nodes/email.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  "attachmentPath": "/mnt/workflows/${tenantId}/evans-pres-repr/evans-pres-repr-${timestamp}.csv",
+  "includeAttachment": "${(count > 0)?then('true', 'false')}",
```
Comment (kaladay) [inline evans-pres-repr/nodes/email.json:26]:
```suggestion
  "includeAttachment": "${(count > 0) ? then('true', 'false')}",
```

File: evans-pres-repr/nodes/end.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "id": "773c1bb9-4222-454d-ace9-413a578aec46",
+  "name": "End Task",
+  "description": "End Task",
```
Comment (kaladay) [inline evans-pres-repr/nodes/end.json:4]: Make this a sentence.

File: evans-pres-repr/nodes/start.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "959c601c-96f9-48a0-847c-6857922ba397",
+  "name": "Start",
+  "description": "",
```
Comment (kaladay) [inline evans-pres-repr/nodes/start.json:4]: Add description.

File: README.md [hunk 11]
```diff
@@ -570,3 +570,32 @@ curl --location --request POST 'http://localhost:9001/mod-workflow/events/workfl
...
+
+To build and activate:
+```shell
+fw build evans-pres-repr
+fw activate evans-pres-repr
+```
+
+Either wait for scheduled event to occur or manually execute via:
+```shell
+fw run evans-pres-repr
```
Comment (kaladay) [inline README.md:600]: Please follow the practice of adding the table, such as is done here:
- https://github.com/TAMULib/fw-registry/blob/59e0027ceb7aa65cc700e8e81e1df7050012c8f9/README.md#books-call-number

File: evans-pres-repr/nodes/disconnect.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+  "id": "53f0e0f0-708e-49f1-b2f0-3b35b7fb6b13",
+  "name": "Disconnect LDP",
+  "description": "",
```
Comment (kaladay) [inline evans-pres-repr/nodes/disconnect.json:4]: Add description.

**Review by kaladay — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rmathew1011) [reply README.md:578]: Removed extra space and improved grammar.

File: evans-pres-repr/nodes/start.json (hunk 10 above)
Comment (rmathew1011) [reply evans-pres-repr/nodes/start.json:4]: Description Added.

File: evans-pres-repr/nodes/end.json (hunk 9 above)
Comment (rmathew1011) [reply evans-pres-repr/nodes/end.json:4]: Made a sentence.

File: evans-pres-repr/nodes/email.json (hunk 8 above)
Comment (rmathew1011) [reply evans-pres-repr/nodes/email.json:26]: Added space before and after '?'

File: evans-pres-repr/nodes/email.json (hunk 7 above)
Comment (rmathew1011) [reply evans-pres-repr/nodes/email.json:24]: Added space before and after '?'

File: evans-pres-repr/nodes/connect.json (hunk 6 above)
Comment (rmathew1011) [reply evans-pres-repr/nodes/connect.json:4]: Description added.

File: evans-pres-repr/nodes/delete.json (hunk 5 above)
Comment (rmathew1011) [reply evans-pres-repr/nodes/delete.json:4]: Description added.

**Review by qtamu — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -570,3 +570,44 @@ curl --location --request POST 'http://localhost:9001/mod-workflow/events/workfl
...
+
+### Evans Pres Repr Workflow (Scheduled)
+
+This workflow sends a monthly email containing a list of all items with 'temporary location' set to "Eva Pres Repr" to a specifically configured email address `evansPresReprFrom`.
+
+These variables are required when triggering the workflow:
+
+| Variable Name           | Allowed Values | Short Description |
+| --------------          | -------------- | ----------------- |
+| path                    | directory path | The directory on the system where the files, like the CSV file, are stored within on the server and contain the `tenantPath` (include trailing slash after the directory). |
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply README.md]: Removed path variable as we have the path value hard coded in the database query delegate task.

**Review by kaladay — APPROVED** [review]

**Pull request merged**